ns1/ipam.py

```diff
@@ -1050,12 +1053,45 @@ def update(self, address_family, options, server_options=None):
             self.server_options = server_options
 
         self.address_family = address_family
-        self.__dict__.update(options)
+        self.options = options
         self.__dict__.update(server_options)
+
         self.option_list = {
             "options": [
-                {"name": "%s/%s" % (self.address_family, key), "value": value}
-                for key, value in options.items()
+                option.generate_option(address_family)
+                for option in self.options
             ]
         }
+
         self.option_list.update(self.server_options)
+
+
+class DHCPOptionValue:
+    def __init__(self, key, value, always_send=None):
+        """
+        Create the DHCPOptionValue class that can be used as value with :class:`ns1.ipam.DHCPOptions`
+
+        :param key str option name
+        :param value any option value
+        :param always_send bool indicates whether this option be sent back in lease or not
+        """
+        self.key = key
+        self.value = value
+        self.always_send = (
+            bool(always_send) if always_send is not None else always_send
+        )
```

Comment:
Usually, inside Behave tests data for always_send field will be the type of string. I think the better way to convert in here rather than force devs to convert to bool before calling this function.

Reply:
self.always_send = bool(always_send)

Reply:
Or `self.always_send = always_send is not None` if you only want to allow None as your falsy value.

Reply:
If the variable is going to end up as a bool, the it’s common for function arguments to take the value as a bool. This is an SDK so it should adhere to that expectation. The burden should be on the user to convert strings, or whatever type, to bool. It’s especially tricky with strings because any non-empty string, including “False”, evaluates to True, whereas empty string and None evaluate to False.

Reply:
With regard to whether `None` should be treated as false in this case an interesting question.  If `always_send` is not specified at the API, then we don't pass it down to kea and then the default behavior -- which is `always_send` is False -- takes over.  So we could use `None` here to mean, "leave the system to its default behavior".  I assumed this was the desired behavior, but I don't have a strong opinion either way.

Reply:
We need NONE here as empty value to suppress this field at all. I that case Kea will use its defaults.

Reply:
Ok. Then with the suggested changes that will work.

Reply:
In my opinion, without any checks with a simple casting value to _bool_ we will have more stability rather than leave this action for the user. 
And yes, @cbertinato, with bool casting it will be tricky anyway, will it be on the user side or here. I don't want to create a hard switching logic for that. Maybe we can changes the type of that field from bool to any in function documentation and add a proper description that will tell a user about _bool_ casting inside?

Reply:
This is externally facing code and it is not common for functions that take bool arguments to take them as strings, at least not in the python world.  I don't think there's anything wrong with putting this responsibility on the user.  It simply doesn't make sense to expose a variable that can only take two values - True and False - as a string.

Reply:
this is good catch - since we're validating later it seems fine not to coerce here, and we won't have things like `"False"` sending `True` to the API

Reply:
Bool casting was placed on the user
Fixed